packages/gatsby-theme-docs/src/components/child-sections-nav.js

```diff
@@ -161,16 +170,47 @@ const ChildSectionsNav = (props) => {
     findCurrentSection(props.parent, pageData.tableOfContents)
   );
 
-  const filteredSectionToCList = sectionToC.items.filter((item) => {
+  const filteredSectionToCList = [];
+  sectionToC.items.forEach((item) => {
+    let matchedItems = [];
     const filterItem = item.title.toLowerCase();
     const values = searchValue.toLowerCase().split(/\W+/);
     const results = values.map((term) => {
       if (term === '' || filterItem.includes(term)) {
         return true;
       }
+      if (item.items) {
+        item.items.forEach((subItem) => {
+          const filterItem = subItem.title.toLowerCase();
+          const subResults = values.map((term) => {
+            if (term === '' || filterItem.includes(term)) {
+              return true;
+            }
+            return false;
```

Comment:
Style opionion: This can be expressed much more concisely.  The `||` OR-expression is evaluating to a boolean anyways, you can directly return it.  You would not even need the body braces and the "return" then